projects/core/src/lib/map/state/map.actions.ts

```diff
@@ -46,6 +46,10 @@ export const moveLayerTreeNode = createAction(
   `${mapActionsPrefix} Move Layer Tree Nodes`,
   props<{ nodeId: string; position: 'before' | 'after' | 'inside'; parentId?: string; sibling?: string; isBaseLayerTree?: boolean }>(),
 );
+export const setLayerTreeNodeChildren = createAction(
+  `${mapActionsPrefix} Set Layer Tree Node Children`,
+  props<{ nodeId: string; children: string[]; isBaseLayerTree?: boolean }>(),
```

Comment:
Why not something like `props<{ nodes: Array<{ nodeId: string; children: string[] }>; isBaseLayerTree?: boolean }>()` so you can send a single update action when reading the bookmark

Reply:
There's no clear guidance as to which approach is preferred for actions; I've seen both used. (e.g. `setLayerOpacity`. I adjusted it here, though.